doc/mvf_v4.rst

```diff
@@ -8,12 +8,441 @@ the data is split into chunks, each in its own file, which are loaded
+
+Here ``.`` is used to indicate a sub-namespace, but the actual separator
+is subject to change and one should always use
+:meth:`~.katsdptelstate.telescope_state.TelescopeState.join` to
+construct compound namespace names.
+
+Keys may move between these namespaces without notice. Readers should
+search for keys from the most specific to the least specific appropriate
+namespace (see for example
+:meth:`katdal.datasources.view_capture_stream`).
+
+Where keys contain strings, they might contain either raw bytes (which
+should be decoded as UTF-8) or Unicode text. Readers should be prepared
+to accept either. The goal is to eventually migrate all such fields to
+use text.
+
+:doc:`katsdptelstate <katsdptelstate:index>` stores two types of values:
+immutable "attributes", and "sensors" which are a list of timestamped
+values. In the documentation below, most keys contain attributes, and
+sensors are indicated.
+
+Global metadata
+^^^^^^^^^^^^^^^
+
+A subset of the sensors in the MeerKAT system are stored in the file, in
+the global namespace. Documenting the MeerKAT sensors is beyond the
+scope of this documentation.
+
+The following keys are also stored.
+
+``sdp_config`` (dict)
+    The JSON object used to configure the SDP subarray product. It is
+    not intended to be parsed (the relevant information is intended to
+    be available via other means), but it contains a wealth of
+    debugging information about the streams and connections between
+    them.
+
+``sdp_capture_block_id`` (string) — sensor
+    All capture block IDs seen so far. This should not be confused with
+    ``capture_block_id``, which indicates the default capture block ID
+    that should be consulted when the file is opened without specifying
+    a capture block ID.
+
+``sdp_image_tag`` (string)
+    The Docker image tag for the Docker images forming the realtime SDP
+    capture system. This is the closest thing to a "version number" for
+    the implementation.
+
+``sdp_image_overrides`` (dict)
+    Alternative Docker image tags for specific services within SDP,
+    overriding ``sdp_image_tag``. Overriding individual images is a
+    debugging tool and it should *always* be empty for science
+    observations.
+
+``config.*`` (dict)
+    Command-line options passed to each of the services within SDP.
+
+``sdp_task_details`` (dict)
+    Debug information about each of the services launched for the
+    subarray product, including the host on which it ran and the Mesos
+    TaskInfo structure.
+
+
+Common stream metadata
+^^^^^^^^^^^^^^^^^^^^^^
+The list of streams that can be accessed from the archive is available
+in ``sdp_archived_streams`` (in the global namespace). Within each
+stream, the following keys may be defined (not all make sense for
+every stream type).
+
+Only ``stream_type`` and ``src_streams`` are guaranteed to be in the
+stream namespace i.e., independent of the capture block. The others may
```

Comment:
I agree there should be a comma before-hand. According to [this](https://www.dailywritingtips.com/comma-after-i-e-and-e-g/), the comma afterwards is a matter of taste/style but more common in American English. I've removed it since we're favouring British English.